src/components/ControlBar.tsx

```diff
@@ -1,115 +1,62 @@
-};
-type PortDefinition = {id: string, name: string};
-type PortState = {
-  availableOutputs: Array<PortDefinition>;
-  availableInputs: Array<PortDefinition>;
-};
-
-class MidiPort extends React.Component<PortProps, PortState> {
-  constructor(props: PortProps) {
-    super(props);
-    const webMidiInitialized = this.props.webMidiStatus === "initialized";
-    const availableInputs = webMidiInitialized ? this.generateAvailableInputs() : [];
-    const availableOutputs = webMidiInitialized ? this.generateAvailableOutputs() : [];
-    this.state = {
-      availableOutputs: availableOutputs,
-      availableInputs: availableInputs,
-    };
-    //this.onSelectedInput = this.onSelectedInput.bind(this);
-  }
-
-  // TODO: Refresh on new device plugged in
-
-  componentDidUpdate(prevProps: PortProps): void {
-    if (prevProps.webMidiStatus === "initializing"){
-      console.log("webMidiStatus initializing")
-      const webMidiInitialized = this.props.webMidiStatus === "initialized";
-      const availableInputs = webMidiInitialized ? this.generateAvailableInputs() : [];
-      const availableOutputs = webMidiInitialized ? this.generateAvailableOutputs() : [];
-      this.setState({availableOutputs, availableInputs});
-    }
-  }
-
-  generateAvailableOutputs(): Array<PortDefinition> {
+export const MidiPort = ({onSelectedInput, webMidiStatus, access} : PortProps) => {
+  const webMidiInitialized = webMidiStatus === "initialized";
+  const [availableInputs, setAvailableInputs] = useState<Array<PortDefinition>>([])
+  const [availableOutputs, setAvailableOutputs] = useState<Array<PortDefinition>>([])
+  
+  function generateAvailableOutputs(): Array<PortDefinition> {
     let outputs = []
     for (const output of WebMidi.outputs) {
       outputs.push({"id": output.id, "name": output.name});
     }
     return outputs;
   }
 
-  generateAvailableInputs(): Array<PortDefinition> {
+  function generateAvailableInputs(): Array<PortDefinition>{
     let inputs = []
     for (const input of WebMidi.inputs) {
       inputs.push({"id": input.id, "name": input.name});
     }
     return inputs;
   }
 
-  componentDidMount() {
-    // this.generatePort(this.props.access);
-  }
-
-  setInputEventHandler(midiAccess: WebMidi.MIDIAccess, midiId: string) {
-    let selectedInput = midiAccess.inputs.get(midiId);
-    selectedInput!.onmidimessage = this.onMIDIMessage;
-  }
-
-  onMIDIMessage(e: WebMidi.MIDIMessageEvent) {
-    this.logMidiInfo(e);
-    //var inputNote = e.data;
-    //var transposedNote = [inputNote[0], inputNote[1] + 12, inputNote[2]];
-  }
-
-  logMidiInfo(e: WebMidi.MIDIMessageEvent) {
-    var str = "MIDI message received at timestamp " + e.timeStamp + "[" + e.data.length + " bytes]: ";
-    for (var i = 0; i < e.data.length; i++) {
-      str += e.data[i].toString(10) + " ";
+  useEffect(() => {
+    console.log(webMidiStatus)
+    if (webMidiInitialized) {
+      setAvailableInputs(generateAvailableInputs())
```

Comment:
Here the updater is already written :)

```suggestion
      setAvailableInputs(generateAvailableInputs)
```